tests/framework/helpers/HtmlTest.php

```diff
@@ -99,7 +100,48 @@ public function testJsFile()
         $this->assertEquals("<!--[if IE 9]>\n" . '<script src="http://example.com"></script>' . "\n<![endif]-->", Html::jsFile('http://example.com', ['condition' => 'IE 9']));
         $this->assertEquals("<!--[if (gte IE 9)|(!IE)]><!-->\n" . '<script src="http://example.com"></script>' . "\n<!--<![endif]-->", Html::jsFile('http://example.com', ['condition' => '(gte IE 9)|(!IE)']));
     }
+    
+    public function testCsrfMetaTagsDisableCsrfValidation()
+    {
+        $this->assertEquals('', Html::csrfMetaTags());
+    }
+    
+    public function testCsrfMetaTagsEnableCsrfValidation()
+    {
+        $this->mockApplication([
+            'components' => [
+                'request' => [
+                    'class' => 'yii\web\Request',
+                    'enableCsrfValidation' => true,
+                    'cookieValidationKey' => 'key',
+                ],
+                'response' => [
+                    'class' => 'yii\web\Response',
+                ],
+            ],
+        ]);
+        $pattern = '<meta name="csrf-param" content="_csrf">%A<meta name="csrf-token" content="%s">';
+        $actual = Html::csrfMetaTags();
+        $this->assertStringMatchesFormat($pattern, $actual);
+    }
+    
+    public function testCsrfMetaTagsEnableCsrfValidationWithoutCookieValidationKey()
+    {
+        $request = $this->getMock('yii\\web\\Request');
+        $request->method('enableCsrfValidation')->willReturn(true);
+        Yii::$app->set('request', $request);
+        $pattern = '<meta name="csrf-param" content="_csrf">%A<meta name="csrf-token">';
+        $actual = Html::csrfMetaTags();
+        $this->assertStringMatchesFormat($pattern, $actual);
+    }
 
+    public function testBeginFormSimulateViaPost()
```

Comment:
Perhaps it https://github.com/yiisoft/yii2/blob/master/framework/helpers/BaseHtml.php#L336 should be so `if (strcasecmp($method, 'get') or strcasecmp($method, 'post')) {`